type_inference/type_retrieval_exception.py

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/python
+#
+# Copyright 2023 Logica Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from logging import error
+from common import color
+
+
+class TypeRetrievalException(Exception):
```

Comment:
The error is very narrow. What if we need raise another exception? Let's keep only the [ Error ] part here and move the message to the place where we raise the exception.

Reply:
I think it's problem with exception name. Proper one is BadSchemaException.
I don't want to pass all exception text everywhere this exception is used